lib/src/clojure_lsp/refactor/transform.clj

```diff
@@ -453,6 +453,32 @@
            {:loc   expr-edit
             :range expr-meta}])))))
 
+(defn ^:private extract-definition-params [zloc]
+  ;; the expression that will be extracted
+  (when-let [zloc (or (z/skip-whitespace z/right zloc)
+                      (z/skip-whitespace z/up zloc))]
+    ;; the top-level form it will be extracted from (possibly the same as zloc)
+    (when-let [form-loc (edit/to-top zloc)]
+      {:zloc zloc
+       :form-loc form-loc})))
+
+(defn can-extract-definition? [zloc]
+  (boolean (extract-definition-params zloc)))
+
+(defn extract-definition [zloc def-name]
+  (when-let [{:keys [zloc form-loc]} (extract-definition-params zloc)]
+    (let [expr-node (z/node zloc)
+          expr-meta (meta expr-node)
+
+          def-name (or def-name "new-value")
+          expr-edit (z/of-node (symbol def-name))
+          def-loc (-> (format "(def ^:private %s\n  )" def-name)
```

Comment:
IIRC extract function checks for the `:use-metadata-for-privacy?` setting which is actually disable by default, I think we should check of that and use `def-` if not enabled instead

Reply:
I thought that too at first, but then I remembered—there's no such thing as `def-`!

Reply:
Oh, great point 😂